wandb/sdk/wandb_run.py

```diff
@@ -3701,6 +3705,65 @@ def pb_to_dict(
                 logger.error("Error getting system metrics: %s", e)
         return {}
 
+    @property
+    @_run_decorator._attach
+    @_run_decorator._noop_on_finish()
+    def _metadata(self) -> Metadata | None:
+        """The metadata associated with this run.
+
+        NOTE: Automatically collected metadata can be overridden by the user.
+        """
+        if self.settings.x_require_legacy_service:
+            wandb.termwarn(
+                "The metadata property is not supported when using the legacy service."
+            )
+            return None
```

Comment:
i was thinking making the change in the internal process side, in `wandb-service` handler/sender, but i don't have a strong preference, just nice that the condition only happens in one place...

Reply:
I didn't want to wire up the messages in service only to make it print a noop warning.

Reply:
ok, up to you...